common/ferrostar/Cargo.toml

```diff
@@ -35,6 +35,7 @@ polyline = "0.11.0"
 rand = "0.9.2"
 serde = { version = "1.0.219", features = ["derive"] }
 serde_json = { version = "1.0.145", default-features = false }
+serde_with = "3.15.1"
```

Comment:
New dependency; this extends some of the classic serde attributes. The one I use in this PR skips serialization for optional fields with a value of `None`. It's possible to do this with `#[serde(skip_serializing_if = "Option::is_none")]` for every single field, but this gets tedious and error prone.